pypesto/sampling/geweke_test.py

```diff
@@ -0,0 +1,204 @@
+import numpy as np
+from scipy.stats import norm
+
+
+def spectrum(x: np.array,
+             nfft: int = None,
+             nw: int = None):
+    """
+    Power spectral density using Hanning window.
+
+    Parameters
+    ----------
+    x:
+        Fraction/fragment of one single parameter
+        of the MCMC chain.
+    nfft:
+        The n-point discrete Fourier Transform.
+    nw:
+        Number of windows.
+
+    Returns
+    -------
+    spectral_density:
+        The spectral density.
+
+    """
+
+    if nfft is None:
+        nfft = np.min(len(x), 256)
+
+    if nw is None:
+        nw = int(nfft / 4)
+
+    n_overlap = int(nw / 2)
+
+    # Hanning window
+    w = .5 * (1 - np.cos(
+        2 * np.pi * np.transpose(np.arange(1, nw+1)) /
+        (nw + 1)))
+    n = len(x)
+    if n < nw:
+        x[nw] = 0
+        n = nw
+
+    # Number of windows
+    k = int((n - n_overlap) / (nw - n_overlap))
+    index = np.arange(nw)
+    # Normalizing scale factor
+    kmu = k * np.linalg.norm(w) ** 2
+    spectral_density = np.zeros((nfft))
+
+    for i in range(k):
+        xw = np.multiply(w, x[index])
```

Comment:
this is pointwise multiplication, right? then just `*` ist fine. for matrix multiplication, `@` is available.